flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/TableConfigOptions.java

```diff
@@ -313,4 +325,32 @@ public InlineElement getDescription() {
             return description;
         }
     }
+
+    /** Strategy to expand columns in {@code SELECT *} queries. */
+    @PublicEvolving
+    public enum ColumnExpansionStrategy implements DescribedEnum {
+        EXCLUDE_ALIASED_VIRTUAL_METADATA_COLUMNS(
+                text(
+                        "Excludes virtual metadata columns that reference a metadata key via an alias. "
+                                + "For example, a column declared as 'c METADATA VIRTUAL FROM k' "
+                                + "is not selected by default if the strategy is applied.")),
+
+        EXCLUDE_DEFAULT_VIRTUAL_METADATA_COLUMNS(
+                text(
+                        "Excludes virtual metadata columns that directly reference a metadata key. "
+                                + "For example, a column declared as 'k METADATA VIRTUAL' "
+                                + "is not selected by default if the strategy is applied."));
```

Comment:
Why would we want to exclude one and not the other? I'm wondering if we need the two values. I'm fine the way it is, just wanted to understand what's the idea here.

Reply:
This is to provide more flexibility. E.g. if a metadata virtual column should serve as a kind of system column, it might make sense to only set `EXCLUDE_DEFAULT_VIRTUAL_METADATA_COLUMNS` and give users the chance to define virtual metadata columns that are not excluded using the `METADATA FROM` clause. Anyway I wanted to give some options already. I was also thinking about supporting exclusions of computed columns in the first version, but this was actually not part of the FLIP.